common/scala/src/main/scala/whisk/core/containerpool/HttpUtils.scala

```diff
@@ -53,10 +58,12 @@ import whisk.core.entity.size.SizeLong
  * @param hostname the host name
  * @param timeout the timeout in msecs to wait for a response
  * @param maxResponse the maximum size in bytes the connection will accept
+ * @param maxConcurrent the maximum number of concurrent requests allowed
  */
-protected[core] class HttpUtils(hostname: String, timeout: FiniteDuration, maxResponse: ByteSize)(
+protected[core] class HttpUtils(hostname: String, timeout: FiniteDuration, maxResponse: ByteSize, maxConcurrent: Int)(
```

Comment:
you could do:

```
-protected[core] class HttpUtils(hostname: String, timeout: FiniteDuration, maxResponse: ByteSize, maxConcurrent: Int)(
-  implicit logging: Logging) {
-  def this(hostname: String, timeout: FiniteDuration, maxResponse: ByteSize)(implicit logging: Logging) =
-    this(hostname: String, timeout: FiniteDuration, maxResponse: ByteSize, 1)
+protected[core] class HttpUtils(hostname: String,
+                                timeout: FiniteDuration,
+                                maxResponse: ByteSize,
+                                maxConcurrent: Int = 1)(implicit logging: Logging) {
+
```

Reply:
Tried this, it fails compilation with an issue in using `object HttpUtils` from elsewhere - doesn't seem right, is there something special when object is public, using private class with default constructor args?